spec/unit/face/config_spec.rb

```diff
@@ -61,20 +61,24 @@ def render(action, result)
     expect(render(:print, result)).to eq("file\n")
   end
 
-  it "prints the section and environment, and not a warning,  when a section is given" do
+  it "prints the section and environment, and not a warning, when a section is given and verbose is set" do
     Puppet.settings.parse_config(<<-CONF)
     [user]
     syslogfacility = file
     CONF
 
+    #This has to be after the settings above, which resets the value
+    Puppet[:log_level] = 'info'
+
     subject.expects(:warn_default_section).never
     subject.expects(:report_section_and_environment).once
 
     result = subject.print("syslogfacility", :section => "user")
     expect(render(:print, result)).to eq("file\n")
   end
 
-  it "prints a warning and the section and environment when no section is given" do
+  it "prints a warning and the section and environment when no section is given and verbose is set" do
+    Puppet::Util::Log.level = 'info'
```

Comment:
How about `Puppet[:log_level] = 'info'` here and the one below?

Also in re-reading the existing tests, I'm thinking we should expect `render` to emit the warnings on stderr, instead of expecting the internal/private methods being called (since those are part of the class we're testing). I think you can do that by nesting the `output` matcher:

```ruby
specify {
  expect { 
    expect(render(:print, result)).to eq("file\n")
  }.to output("Resolving settings from section ...").to_stderr
}
```

Reply:
I fixed the log level settings.

Reply:
For the test output, I wish we had a better way to rearrange our architecture.  I agree with you about not testing against internal methods, but I wish I could isolate the IO calls because text matching and relying on output appearance tends to be brittle.